src/Aspire.Hosting.Azure.EventHubs/AzureEventHubsExtensions.cs

```diff
@@ -0,0 +1,101 @@
+    /// </summary>
+    /// <param name="builder">The builder for the distributed application.</param>
+    /// <param name="name">The name of the resource.</param>
+    /// <param name="configureResource">Optional callback to configure the Event Hubs namespace.</param>
+    /// <returns></returns>
+    [Experimental("ASPIRE0001", UrlFormat = "https://aka.ms/dotnet/aspire/diagnostics#{0}")]
+    public static IResourceBuilder<AzureEventHubsResource> AddAzureEventHubs(this IDistributedApplicationBuilder builder, string name,
+        Action<IResourceBuilder<AzureEventHubsResource>, ResourceModuleConstruct, EventHubsNamespace>? configureResource)
+    {
+        var configureConstruct = (ResourceModuleConstruct construct) =>
+        {
+            var eventHubsNamespace = new EventHubsNamespace(construct, name: name);
+
+            eventHubsNamespace.Properties.Tags["aspire-resource-name"] = construct.Resource.Name;
+
+            eventHubsNamespace.AssignProperty(p => p.Sku.Name, new Parameter("sku", defaultValue: "Standard"));
+
+            var eventHubsDataOwnerRole = eventHubsNamespace.AssignRole(RoleDefinition.EventHubsDataOwner);
+            eventHubsDataOwnerRole.AssignProperty(p => p.PrincipalType, construct.PrincipalTypeParameter);
+
+            eventHubsNamespace.AddOutput("eventHubsEndpoint", sa => sa.ServiceBusEndpoint);
+
+            var azureResource = (AzureEventHubsResource)construct.Resource;
+            var azureResourceBuilder = builder.CreateResourceBuilder(azureResource);
+            configureResource?.Invoke(azureResourceBuilder, construct, eventHubsNamespace);
+
+            foreach (var hub in azureResource.Hubs)
+            {
+                var hubResource = new EventHub(construct, name: hub.Name, parent: eventHubsNamespace);
+                hub.Configure?.Invoke(azureResourceBuilder, construct, hubResource);
+            }
+
+        };
+        var resource = new AzureEventHubsResource(name, configureConstruct);
+
+        return builder.AddResource(resource)
+                      // These ambient parameters are only available in development time.
+                      .WithParameter(AzureBicepResource.KnownParameters.PrincipalId)
+                      .WithParameter(AzureBicepResource.KnownParameters.PrincipalType)
+                      .WithManifestPublishingCallback(resource.WriteToManifest);
+    }
+
+    /// <summary>
+    /// Adds an Azure Event Hubs hub resource to the application model. This resource requires an <see cref="AzureEventHubsResource"/> to be added to the application model.
+    /// </summary>
+    /// <param name="builder">The Azure Event Hubs resource builder.</param>
+    /// <param name="name">The name of the Event Hub.</param>
+    public static IResourceBuilder<AzureEventHubsResource> AddEventHub(this IResourceBuilder<AzureEventHubsResource> builder, string name)
```

Comment:
This API pattern differs from how databases work. I think it will be surprising to our customers.

In databases, we have:

```C#
var db = builder.AddPostgres("db-server").AddDatabase("db1");

builder.AddProject<Projects.Api>("api")
    .WithReference(db);
```

When executing that code, `db` points to the `db1` resource - a "child" of the `db-server`. So when you `.WithReference(db)`, the connection string name that you use in your app is going to be `db1`.

But here,

```C#
var hub = builder.AddEventHubs("eventhub-ns").AddEventHub("hub");

builder.AddProject<Projects.Api>("api")
    .WithReference(hub);
```

`hub` is going to point to the `eventhub-ns` resource, and not the `hub` resource. So in your web app, you need to use `builder.AddAzureEventHubProducerClient("eventhub-ns")`, which is different than how databases work.

Reply:
Agree 100% -- see my comment above. The service bus resource is equally deficient, and I copied that code. I'd want to see this fixes for both ASB and AEH in P6.  /cc @mitchdenny

Reply:
Does this work with managed identity?

Reply:
> Does this work with managed identity?

Did you mean to put this here, @davidfowl ? If so, I don't see why not?